rand_chacha/src/chacha.rs

```diff
@@ -629,4 +629,29 @@ mod test {
         rng.set_word_pos(0);
         assert_eq!(rng.get_word_pos(), 0);
     }
+
+    #[test]
+    fn test_chacha_next_u32_vs_next_u64() {
+        let mut rng1 = ChaChaRng::from_seed(Default::default());
+        let mut rng2 = rng1.clone();
+        let mut rng3 = rng1.clone();
+
+
+        let mut a = [0; 16];
+        (&mut a[..4]).copy_from_slice(&rng1.next_u32().to_le_bytes());
+        (&mut a[4..12]).copy_from_slice(&rng1.next_u64().to_le_bytes());
+        (&mut a[12..]).copy_from_slice(&rng1.next_u32().to_le_bytes());
+
+        let mut b = [0; 16];
+        (&mut b[..4]).copy_from_slice(&rng2.next_u32().to_le_bytes());
+        (&mut b[4..8]).copy_from_slice(&rng2.next_u32().to_le_bytes());
+        (&mut b[8..]).copy_from_slice(&rng2.next_u64().to_le_bytes());
+        assert_eq!(a, b);
+
+        let mut c = [0; 16];
+        (&mut c[..8]).copy_from_slice(&rng3.next_u64().to_le_bytes());
+        (&mut c[8..12]).copy_from_slice(&rng3.next_u32().to_le_bytes());
+        (&mut c[12..]).copy_from_slice(&rng3.next_u32().to_le_bytes());
+        assert_eq!(a, b);
```

Comment:
I would expect `rng2` and `rng3` to produce the same result, but not `rng1` (which would discard 32-bits).

Also, last line should test `c`.

Reply:
*Also* the code in question pertains to `BlockRngCore<Item = u64>`, which IIRC is only used by `Issac64`.

Reply:
@dhardy I think the test is fine for `BlockRng`. Do you think I should remove it, or can it stay as it is?